### PR TITLE
jenkins: peer constraints

### PR DESCRIPTION
**Pull request by egonelbre**
Check whether any of the libraries import any of the peers. It's easy to introduce such a dependency right now.

Currently doesn't fail the linter since we still have violations in our code.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: scripts/check-peer-constraints.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	var buildFlags []string
+	if *race {
+		buildFlags = append(buildFlags, "-race")
+	}
+
+	return packages.Load(&packages.Config{
+		Mode:       packages.LoadImports,
+		Env:        os.Environ(),
+		BuildFlags: buildFlags,
+		Tests:      false,
```
Comment (zeebo) [inline scripts/check-peer-constraints.go]: I think we might want to include tests as well.

File: scripts/check-peer-constraints.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			}
+			visit(imp, append(path, pkg))
+		}
+	}
+
+	for _, pkg := range source {
+		visit(pkg, nil)
+	}
+
+	return links
```
Comment (zeebo) [inline scripts/check-peer-constraints.go]: when is links ever set to true?

File: scripts/check-peer-constraints.go (hunk 1 above)
Comment (egonelbre) [reply scripts/check-peer-constraints.go]: We a have a lot of system level tests. To test uplink fully we need to have a running storage node and satellite.

File: scripts/check-peer-constraints.go (hunk 2 above)
Comment (egonelbre) [reply scripts/check-peer-constraints.go]: ah, missed setting it to true.

File: scripts/check-peer-constraints.go (hunk 1 above)
Comment (zeebo) [reply scripts/check-peer-constraints.go]: Dang. Works for me. Additionally, if we ever have problems we can iterate on catching them here.

File: scripts/check-peer-constraints.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+func main() {
+	flag.Parse()
+
+	var exitcode int
+
+	peers, err := load(
+		"storj.io/storj/satellite/...",
+		"storj.io/storj/storagenode/...",
+		"storj.io/storj/bootstrap/...",
```
Comment (stefanbenten) [inline scripts/check-peer-constraints.go]: Should we add the versioncontrol here as well?
It doesnt have any sub packages yet, but that could happen at any stage

Comment (egonelbre) [reply scripts/check-peer-constraints.go]: added

**Pull request merged**